Correction of Bug in HW Baseboard Get Input Routine

RKStrand opened the pull request:
Pull request overview
---------------------
 - Fixes #9742 
 - A user noticed that changes to the fraction of radiant energy to people in the radiant/convective hot water baseboard unit resulted in large changes to the energy consumption of the system that was not present previously.  In an older version, changes to this parameter from 0.3 to 0.1 resulted in a reasonable 1% difference in the results.  In the latest version of EnergyPlus, the change was nearly 15%.  This was traced back to an error in the way the fraction of heat added via convection to the zone was being calculated and was introduced when the baseboard input object was split into two part--basic and design parameters.  As a result, the baseboard unit was not conserving energy.  This is now corrected and the differences are back to what they were before and should be (roughly 1%).  Along with a unit test to verify that this is now working properly, one of the test suite input files was altered to specifically examine at the situation that caused the problem.  So, in the ci results, that specific file does show differences as expected.  No other files show differences, even in the files that contain radiant convective hot water baseboard units because in all of those other files the fraction radiant and the fraction of radiant to people were both set to 0.3, masking the problem.  The other radiant convective baseboard units were also checked and they do not have the same problem that the hot water radiant convective baseboard unit had.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Correction of Defect 9742

When the various radiant and radiant/convective systems were split into base and design input objects, a very subtle error was introduced.  This corrects the problem.

RKStrand committed:
Unit Test for 9742

New unit test for Defect 9742 and modified input file to test the problem that was noticed by a user.

RKStrand committed:
Fix format error in compiler check

See title.

rraustad commented on src/EnergyPlus/HWBaseboardRadiator.cc:
```diff
@@ -605,7 +605,7 @@ namespace HWBaseboardRadiator {
                 thisHWBaseboard.FracConvect = 0.0;
             } else {
-                thisHWBaseboard.FracConvect = 1.0 - AllFracsSummed;
+                thisHWBaseboard.FracConvect = 1.0 - HWBaseboardDesignDataObject.FracRadiant;
```
You no longer need line 598 since AllFracsSummed is also set at line 636. I see this same code in ChilledCeilingPanelSimple so what is the plan for other objects?

rraustad replied:
Correction: the code in ChilledCeilingPanelSimple does use FracRadiant in this part of the code.

RKStrand replied:
@rraustad I can get rid of line 598, not a problem.  I did check on the other radiant convective baseboard units and these were correct.  It seems like this was a one-off issue when the baseboard object was split into two pieces.

RKStrand replied:
@rraustad Got rid of line 598 and standardized all of the models that use this code in the latest commit.

RKStrand committed:
Clean up of AllFracsSummed

Cleaned up AllFracsSummed variable, eliminating where it wasn't being used for anything and moving the declarations to the local portion of the Get routines.

rraustad commented on tst/EnergyPlus/unit/HWBaseboardRadiator.unit.cc:
```diff
@@ -190,3 +194,123 @@ TEST_F(EnergyPlusFixture, HWBaseboardRadiator_HWBaseboardWaterFlowResetTest)
+    ASSERT_NEAR(state->dataHWBaseboardRad->HWBaseboardDesignObject(1).FracRadiant, 0.4, absTol);
+    ASSERT_NEAR(state->dataHWBaseboardRad->HWBaseboardDesignObject(1).FracDistribPerson, 0.2, absTol);
+    ASSERT_NEAR(state->dataHWBaseboardRad->HWBaseboard(1).FracConvect, 0.6, absTol);
+}
```
Unit test looks OK.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 9742RadConvBaseboardPercentRadiantBug

mjwitte reviewed (approved):
Tested defect file with 0.3 and 0.1, got expected changes in results. Updated the branch, will let CI run one more round.

mjwitte commented:
CI results are as expected, all green except diffs in one file. Merging.
